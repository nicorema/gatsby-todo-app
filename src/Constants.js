export const BREAKPOINTS = {
  sm: '640px',
  md: '768px',
  lg: '1024px',
  xl: '1280px',
}

export const BOGOTA_CENTER = {
  lat: 4.624335,
  lng: -74.063644,
}

export const NAVIGATION_LINKS = [
  { link: '/todos-manager', title: 'Todos Manager' },
  { link: '/todos-log', title: 'Todos Log' },
]
