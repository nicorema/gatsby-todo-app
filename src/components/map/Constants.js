export const MAX_TOOLTIP_CHARACTERS = 50
