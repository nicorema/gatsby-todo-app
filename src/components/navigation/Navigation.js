import React from 'react'
import { Link } from 'gatsby-plugin-intl'
import { StyledLink, StyledNav } from './Navigation.styles'
import { NAVIGATION_LINKS } from '../../Constants'

const Navigation = () => {
  return (
    <StyledNav>
      <ul>
        {NAVIGATION_LINKS.map(({ link, title }) => (
          <li key={`${link}-${title}`}>
            <StyledLink>
              <Link to={link}>{title}</Link>
            </StyledLink>
          </li>
        ))}
      </ul>
    </StyledNav>
  )
}

export default Navigation
