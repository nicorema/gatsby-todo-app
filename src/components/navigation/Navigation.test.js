import React from 'react'
import renderer from 'react-test-renderer'
import { NAVIGATION_LINKS } from '../../Constants'

import Navigation from './Navigation'

describe('Navigation', () => {
  const navigationCmp = renderer.create(<Navigation />)

  it('renders correctly', () => {
    const snapShot = navigationCmp.toJSON()
    expect(snapShot).toMatchSnapshot()
  })

  it('renders correct ammount of links', () => {
    const listItems = navigationCmp.root.findAllByType('li')
    expect(listItems.length === NAVIGATION_LINKS.length)
  })
})
