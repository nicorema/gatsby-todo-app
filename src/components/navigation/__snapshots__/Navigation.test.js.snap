// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  className="sc-bdnylx giuwNz"
>
  <ul>
    <li>
      <span
        className="css-qyfojd"
      >
        <a
          href="/todos-manager"
        >
          Todos Manager
        </a>
      </span>
    </li>
    <li>
      <span
        className="css-qyfojd"
      >
        <a
          href="/todos-log"
        >
          Todos Log
        </a>
      </span>
    </li>
  </ul>
</nav>
`;
